src/whisper_at_train/run_combined_training.sh

```
@@ -7,11 +7,22 @@
-source /home/taemyung_heo/.cache/pypoetry/virtualenvs/whisper-at-z6hdRBdT-py3.10/bin/activate
-export TORCH_HOME=../../pretrained_models
+set -euxo pipefail
+
+# Environment variables with defaults
+WHISPER_AT_VENV="${WHISPER_AT_VENV:-/home/taemyung_heo/.cache/pypoetry/virtualenvs/whisper-at-z6hdRBdT-py3.10/bin/activate}"
+WHISPER_AT_BASE="${WHISPER_AT_BASE:-$(dirname $(dirname $(pwd)))}"
+TORCH_HOME="${TORCH_HOME:-${WHISPER_AT_BASE}/pretrained_models}"
+
+# Activate virtual environment
+if [ -f "${WHISPER_AT_VENV}" ]; then
+    source "${WHISPER_AT_VENV}"
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)